### PR TITLE
Use valid hookFor signature (Fix #28)

### LINKED ISSUE #28: Sub-generators (component, mixin) run `all` generator
If you try to run `yo flight:component` then it will actually invoke `yo flight:all`. @passy, have you guys made any changes to `yo` that could have introduced this bug? This generator hasn't changed for a while and used to work. Any idea how to fix this?

### PR DESCRIPTION
**Pull request by SBoudrias**
From what I see from the source code managing `hookFor`, it never allowed using subgenerator in its name directly. If it missed a context `as` value, it then appended the current generator path to the string...

This wasn't causing issue before because we recently refactored the lookup and generator resolution logic.

Anyway, this will fix it for now. And we'll fix it upstream in the generator system for next release.

### CODE REVIEW

**Comment by necolas** [comment]
Thanks! I suspected it might be the `hookFor` code. Appreciate you helping us with this.

File: lib/generators/all/index.js [hunk 1]
```diff
@@ -33,11 +33,13 @@ function Generator() {
     }
   });
 
-  this.hookFor('flight:component', {
+  this.hookFor('flight', {
+    as: 'component',
     args: ['my_component']
```
Comment (necolas) [inline lib/generators/all/index.js:38]: This argument doesn't seem to be getting passed anymore. The generated 'component' is now called "my_mixin", so it seems to be getting the argument from the next `hookFor`? Not sure what is going on.

Comment (necolas) [inline lib/generators/all/index.js:38]: Fixed by upgrading to yeoman-generator 0.16.0

Comment (SBoudrias) [inline lib/generators/all/index.js:38]: Ok! I'm happy to know it've been fixed :)

**Pull request merged**